hpOneView/resources/resource.py

```diff
@@ -186,7 +186,7 @@ def delete(self, resource, force=False, timeout=-1, custom_headers=None):
 
         task, body = self._connection.delete(uri, custom_headers=custom_headers)
 
-        if not task:
+        if not task or 'taskState' not in task:
```

Comment:
I suggest you to put this validation inside task_monitor.wait_for_task. The task monitor should deal properly when not receiving a task.